sympy/assumptions/assume.py

```diff
@@ -86,80 +132,226 @@ def arg(self):
         x + 1
 
         """
-        return self._args[1]
+        # Will be deprecated
+        args = self._args
+        if len(args) == 2:
+            # backwards compatibility
+            return args[1]
+        raise TypeError("'arg' property is allowed only for unary predicate.")
 
     @property
     def args(self):
+        # Will be deprecated and return normal Basic.func
         return self._args[1:]
 
     @property
     def func(self):
+        # Will be deprecated and return normal Basic.func
         return self._args[0]
 
-    @cacheit
-    def sort_key(self, order=None):
-        return (self.class_key(), (2, (self.func.name, self.arg.sort_key())),
-                S.One.sort_key(), S.One)
-
-    def __eq__(self, other):
-        if type(other) is AppliedPredicate:
-            return self._args == other._args
-        return False
+    @property
+    def function(self):
+        # Will be changed to self.args[0] after args overridding is removed
+        return self._args[0]
 
-    def __hash__(self):
-        return super().__hash__()
+    @property
+    def arguments(self):
+        # Will be changed to self.args[1:] after args overridding is removed
+        return self._args[1:]
 
     def _eval_ask(self, assumptions):
-        return self.func.eval(self.arg, assumptions)
+        return self.function.eval(self.arguments, assumptions)
 
     @property
     def binary_symbols(self):
         from sympy.core.relational import Eq, Ne
-        if self.func.name in ['is_true', 'is_false']:
-            i = self.arg
+        from .ask import Q
+        if self.function == Q.is_true:
+            i = self.arguments[0]
             if i.is_Boolean or i.is_Symbol or isinstance(i, (Eq, Ne)):
                 return i.binary_symbols
         return set()
 
 
-class Predicate(Boolean):
+class PredicateMeta(ManagedProperties):
+    """
+    Metaclass for ``Predicate``
+
+    If class attribute ``handler`` is not defined, assigns empty Dispatcher
+    to it.
     """
-    A predicate is a function that returns a boolean value.
+    def __new__(cls, clsname, bases, dct):
+        if "handler" not in dct:
+            name = ''.join(["Ask", clsname.capitalize(), "Handler"])
```

Comment:
I think 3.6 is allowed in sympy now:
```suggestion
            name = f"Ask{clsname.capitalize()}Handler"
```

Reply:
Yes, Python 3.5 is no longer supported as of sympy 1.7.1.